File: yeji/DP/counting-bits.js
var countBits = function(n) {
    let dp = [0]
    let offset = 1;
    
    for (let i=1; i< n+1; i++) {
        dp.push(0)
    }
    
    
    for (let i=1; i < n+1; i++) {
        if ((offset * 2) === i) {
            offset = i
        }
        dp[i] = dp[i-offset] + 1
    }
    
    return dp
};

File: yeji/DP/counting-bits.py
class Solution(object):
    def countBits(self, n):        
        dp = [0] * (n+1)

        for i in range(1, n+1):
            share = i // 2
            if i%2 == 0:
                dp[i] = dp[share]
            else:
                dp[i] = dp[share] + 1
        
    
        return dp
"""
다른 사람 풀이

class Solution(object):
    def countBits(self, n):
        dp = [0] * (n+1)
        offset = 1
        
        for i in range(1, n+1):
            if offset * 2 == i:
                offset = i
            dp[i] = dp[i-offset] + 1
            
        return dp
"""